sdk/service_test.go

```diff
@@ -38,3 +39,34 @@ func Test_FormatServiceName(t *testing.T) {
 		}
 	}
 }
+
+func Test_FormatGatewaySwarm(t *testing.T) {
+	tests := []struct {
+		title       string
+		URL         string
+		suffix      string
+		expectedURL string
+	}{
+		{
+			title:       "URL formatted for Swarm",
```

Comment:
Now do a test where there is no port i.e. `8080` is not given.

Reply:
It will fail if the port is not given.I though the format should always be with port for internal network?

Reply:
It won't fail if a port is not given, remember that we have more than one service we have to resolve, not just the gateway. If no port is given http:// means port 80 and https:// means port 443